truss/parsesvcname/parsesvcname.go

```diff
@@ -0,0 +1,87 @@
+// Package parsesvcname will parse the service name of a protobuf package. The name returned will always be camelcased.
+package parsesvcname
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/TuneLab/go-truss/svcdef"
+	"github.com/TuneLab/go-truss/truss/execprotoc"
+	"github.com/pkg/errors"
+)
+
+// FromPaths accepts the paths of protobuf definition files and returns the
+// name of the service in that protobuf definition file.
+func FromPaths(gopath []string, protoDefPaths []string) (string, error) {
```

Comment:
So after reading how this works I feel like it might be simpler to just modify `cmd/truss/main.go` to: 
- Start by generating the .pb.go files to either the specified `pbout` flag, or to a temporary directory. 
- With the .pb.go files, create a `svcdef` from the `.pb.go` files. 
    - Either from the temp dir, or from `pbout` flag.
- With the service name
    - Create that service directory
    - Or if a temp dir was used, then create the service directory and move all contents of the temp dir to the service directory.

This way all the files are not generated twice and we also don't create an `svcdef` twice. 

Though since truss is not a long running application, and it is already really fast, maybe this reorganization is an over optimization. And it is fine to do all this "heavy stuff" twice to make the flow from a high level clearer. Though for anything high performance or long running this would not be good, so it strikes me as a bad habit.

@tunezaq , @hasIan , I would like your opinions on if my idea is an over optimization.